src/ui/app/jsx/event-screen.jsx

```diff
@@ -0,0 +1,201 @@
+    // event-list
+    diagnosticEvents: React.PropTypes.object.isRequired,
+    listenSubscriptionSubject: React.PropTypes.object.isRequired,
+  },
+
+  getInitialState: function() {
+    return {activeSubscription: null, activeEventSource: null};
+  },
+
+  componentWillMount: function() {
+
+    this._listenSubscription = this.props.listenSubscriptionSubject.subscribeOnNext(subscription => {
+
+      if(this.state.activeEventSource) {
+        this.state.activeEventSource.close();
+      }
+
+      const URL_PREFIX = getUrlPrefix(window.top.location.pathname);
+      const source = new EventSource(`${URL_PREFIX}/diag_event/sse_listen/${subscription.id}`);
+      const diagnosticEvents = this.props.diagnosticEvents;
+      source.onmessage = function(event) {
+        if(event.data == "PING") {
+          return;
+        }
+        const obj = $.parseJSON(event.data);
+        console.debug(`Received diagnostic event (${event.lastEventId})`, obj);
+        diagnosticEvents.onNext({eventData: obj, eventId: event.lastEventId});
+      };
+      source.onerror = function(err) {
+        diagnosticEvents.onError(err);
+      };
+      source.onopen = function() {
+        console.debug("EventSource ready for receiving diagnostic events");
+      };
+
+      this.setState({activeSubscription: subscription, activeEventSource: source});
+    });
+
+    this._unlistenSubscription = this.props.unlistenSubscriptionSubject.subscribeOnNext(subscription => {
+      if(this.state.activeSubscription) {
+        if(this.state.activeEventSource) {
+          this.state.activeEventSource.close();
+        }
+        this.setState({activeSubscription: null, activeEventSource: null});
+      }
+    });
+
+    this._deleteSubscription = this.props.deleteSubscriptionSubject.subscribeOnNext(subscription => {
+      if(this.state.activeSubscription && this.state.activeSubscription.id == subscription.id) {
+        if(this.state.activeEventSource) {
+          this.state.activeEventSource.close();
+        }
+        this.setState({activeSubscription: null, activeEventSource: null});
+      }
+    });
+
+    this._clusterSelectedSubscription = this.props.clusterSelected.subscribeOnNext(this._onCancelView);
+  },
+
+  componentWillUnmount: function() {
+    this._listenSubscription.dispose();
+    this._unlistenSubscription.dispose();
+    this._deleteSubscription.dispose();
+  },
+
+  _onCancelView: function() {
+    if(this.state.activeSubscription) {
+      this.props.unlistenSubscriptionSubject.onNext(this.state.activeSubscription);
+    }
+  },
+
+  render: function() {
+
+    var btnLiveView = <button type="button" className="btn btn-sm btn-block btn-default" disabled="disabled">No active subscriptions for live view</button>
+    if (this.state.activeSubscription) {
+      btnLiveView = <button type="button" className="btn btn-sm btn-block btn-success" onClick={this._onCancelView}>Viewing events for {this.state.activeSubscription.description}</button>
```

Comment:
I really don't understand this choice (not meaning to be rude here ;) ).
Nothing indicates that pressing the button will stop the flow of events, and using a green button doesn't show that you're going to stop something.
I really vote in favor of an orange button and a text that says "Press to stop..."

Reply:
UI/UX is tricky.

I never interpreted it as a button. And never read its text as an action, but rather as state.
To toggle it to disable it was intuitive enough. And displayed as green indicated to me that it was "active". If it were orange on the other hand I might think that there was something wrong and it wasn't properly activated.

This is a beta feature at the moment, and I'm happy to let it in as the contributor designed it, and to come back on it when broader feedback is received on it.
@spodkowinski what are your thoughts on this?

Reply:
Red elements should point users to issues that need attention. Having the event flow active is the normal state, so we're green. It's also perfectly safe to click the green button. Having the flow paused may be temporary, but should be highlighted as such to get attention, so it's red. You're clicking to get back to normal again (green).

But this is all just my personal, totally unscientific view on this and I'm not considering myself an UX expert.